realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -521,6 +521,99 @@ private void checkEmpty(String str) {
     }
 
     /**
+     * Returns the column indices for the given field name. If a linked field is defined, the column index for
+     * each field is returned
+     *
+     * @param fieldDescription fieldName or link path to a field name.
+     * @param validColumnTypes Legal field type for the last field a
+     * @return list of column indices.
+     */
+    // TODO: consider another caching strategy so linked classes are included in the cache.
+    long[] getColumnIndices(String fieldDescription, RealmFieldType... validColumnTypes) {
```

Comment:
Varargs... it's probably better to change the signature to (String s, RealmFieldType first, RealmFieldType... others)

http://jtechies.blogspot.dk/2012/07/item-42-use-varargs-judiciously.html